app/services/course_option_check.rb

```diff
@@ -0,0 +1,49 @@
+class CourseOptionCheck
+  class InvalidProviderError < StandardError
+    def message
+      'Invalid provider for provided course option'
+    end
+  end
+  class InvalidCourseError < StandardError
+    def message
+      'Invalid course for provided course option'
+    end
+  end
+  class InvalidStudyModeError < StandardError
+    def message
+      'Invalid study mode for provided course option'
+    end
+  end
+  class InvalidLocationError < StandardError
+    def message
+      'Invalid location for provided course option'
+    end
+  end
+
+  attr_reader :provider_id, :course_id, :course_option_id, :study_mode, :location_id, :recruitment_cycle_year
+
+  def initialize(
+    provider_id:,
+    course_id:,
+    course_option_id:,
+    study_mode:,
+    location_id:,
+    recruitment_cycle_year: RecruitmentCycle.current_year
```

Comment:
we don't seem to be using this attribute?

Reply:
We are indeed not using it. We were thinking that that would ensure that the provided course option will be in the current recruitment cycle but I guess if we go down that route then we need to check if the course is open on apply etc which is a bit excessive I think.